PWGHF/TableProducer/candidateSelectorDstar.cxx

```diff
@@ -0,0 +1,285 @@
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file candidateSelectorDstar.cxx
+/// \brief Reconstruction of D* decay candidates
+///
+/// \author Vít Kučera <vit.kucera@cern.ch>, CERN
+/// \author Deependra Sharma <deependra.sharma@cern.ch>, IITB
+/// \author Fabrizio Grosa <fabrizio.grosa@cern.ch>, CERN
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/runDataProcessing.h"
+#include "Framework/Logger.h"
+
+#include "Framework/O2DatabasePDGPlugin.h"
+
+#include "Common/Core/TrackSelectorPID.h"
+#include "Framework/AnalysisDataModel.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+// using namespace o2::aod::HFCandDStarProng;
+// using namespace o2::aod::hf_cand_2prong;
+using namespace o2::aod;
+// using namespace o2::analysis::hf_cuts_d0_to_pi_k;
+// using namespace o2::analysis::hf_cuts_dstar_to_pi_d0;
+using namespace o2::analysis;
+
+// Struct to applying Dstar selection cuts
+struct HfCandidateSelctorDstar {
+  Service<o2::framework::O2DatabasePDG> o2ServicePDG;
+  Produces<aod::HfSelD0> hfSelD0Candidate;
+  Produces<aod::HfSelDstar> hfSelDstarCandidate;
+
+  struct D0Configurables : ConfigurableGroup {
+    Configurable<double> ptD0CandMin{"ptD0CandMin", 0., "Lower bound of D0 candidate pT"};
+    Configurable<double> ptD0CandMax{"ptD0CandMax", 50., "Upper bound of D0 candidate pT"};
+
+    // topological cuts
+    Configurable<std::vector<double>> binsPtForD0{"binsPtForD0", std::vector<double>{hf_cuts_d0_to_pi_k::vecBinsPt}, "pT bin limits for D0"};
+    Configurable<LabeledArray<double>> cutsForD0{"cutsForD0", {hf_cuts_d0_to_pi_k::cuts[0], hf_cuts_d0_to_pi_k::nBinsPt, hf_cuts_d0_to_pi_k::nCutVars, hf_cuts_d0_to_pi_k::labelsPt, hf_cuts_d0_to_pi_k::labelsCutVar}, "D0 candidate selection per pT bin"};
+
+  } d0configurable;
+
+  struct DstarConfigurable : ConfigurableGroup {
+    Configurable<double> ptDstarCandMin{"ptDstarCandMin", 0., "Lower bound of Dstar candidate pT"};
+    Configurable<double> ptDstarCandmax{"ptDstarCandmax", 50., "upper bound of Dstar candidate pT"};
+
+    // topological cuts
+    Configurable<std::vector<double>> binsPtForDstar{"binsPtForDstar", std::vector<double>{hf_cuts_dstar_to_pi_d0::vecBinsPt}, "pT bin limits for Dstar"};
+    Configurable<LabeledArray<double>> cutsForDstar{"cutsForDstar", {hf_cuts_dstar_to_pi_d0::cuts[0], hf_cuts_dstar_to_pi_d0::nBinsPt, hf_cuts_dstar_to_pi_d0::nCutVars, hf_cuts_dstar_to_pi_d0::labelsPt, hf_cuts_dstar_to_pi_d0::labelsCutVar}, "Dstar candidate selection per pT bin"};
+
+  } dstarconfigurable;
```

Comment:
Why did you use `ConfigurableGroups`? (this is a naive question because I don't know what they are indeed :-))

Reply:
I used ConfigurableGroups so that configurables are segregated on the basis if they belongs to D0 only or Dstar only or common. I thought this way code would look more understandable for new reader/debuger. Should I remove it?

Reply:
I removed these configurable group structures.